pkg/types/provider.go

```diff
@@ -18,4 +24,63 @@ type Plugin = PluginProvider
 type PluginProvider interface {
 	ConfigProvider
 	ContractTransmitter() ocrtypes.ContractTransmitter
+	ChainReader() ChainReader
+}
+
+// General error types for providers to return--can be used to wrap more specific errors.
+// These should work with or without LOOP enabled, to help the client decide how to handle
+// an error. The structure of any wrapped errors would normally be automatically flattened
+// to a single string, making it difficult for the client to respond to different categories
+// of errors in different ways. This lessons the need for doing our own custom parsing of
+// error strings.
+type InvalidArgumentError string
+
+func (e InvalidArgumentError) Error() string {
+	return string(e)
+}
+
+func (e InvalidArgumentError) GRPCStatus() *status.Status {
+	return status.New(codes.InvalidArgument, e.Error())
+}
+
+func (e InvalidArgumentError) Is(target error) bool {
+	if e == target {
+		return true
+	}
+
+	return grpcErrorHasTypeAndMessage(target, string(e), codes.InvalidArgument)
+}
+
+type UnimplementedError string
+
+func (e UnimplementedError) Error() string {
+	return string(e)
+}
+
+func (e UnimplementedError) GRPCStatus() *status.Status {
+	return status.New(codes.Unimplemented, e.Error())
+}
+
+func (e UnimplementedError) Is(target error) bool {
+	if e == target {
+		return true
+	}
+
+	return grpcErrorHasTypeAndMessage(target, string(e), codes.Unimplemented)
+}
+
+func grpcErrorHasTypeAndMessage(target error, msg string, code codes.Code) bool {
+	s, ok := status.FromError(target)
+	if !ok || s.Code() != code {
+		return false
+	}
+
+	errs := strings.Split(s.Message(), ":")
+	for _, err := range errs {
+		if strings.Trim(err, " ") == msg {
+			return true
+		}
+	}
+
+	return false
```

Comment:
Could leverage some new slices api here:
```suggestion
	return slices.ContainsFunc(errs, func(err error) bool {
		return strings.Trim(err, " ") == msg 
	}
```